### PR TITLE
Add `EstimatedComputeUnits` platform property

### PR DESCRIPTION
**Pull request by siggisim**
---

**Version bump**: Patch

### CODE REVIEW

File: enterprise/server/tasksize/tasksize.go [hunk 1]
```diff
@@ -2,13 +2,20 @@ package tasksize
...
 	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
 	scpb "github.com/buildbuddy-io/buildbuddy/proto/scheduler"
 )
 
 const (
 	testSizeEnvVar = "TEST_SIZE"
+	// A BuildBuddy Compute Unit is defined as 1 cpu and 2.5GB of memory.
+	estimatedComputeUnitsPropertyKey = "EstimatedComputeUnits"
+	maxComputeUnits                  = 8         // We can't schedule tasks larger than 8 compute units.
+	computeUnitsToMilliCPU           = 1000      // 1 BPU = 1000 milli-CPU
```
Comment (tylerwilliams) [inline enterprise/server/tasksize/tasksize.go:17]: BCU? (or what is a BPU?)

File: enterprise/server/tasksize/tasksize.go [hunk 2]
```diff
@@ -2,13 +2,20 @@ package tasksize
...
 
 	repb "github.com/buildbuddy-io/buildbuddy/proto/remote_execution"
 	scpb "github.com/buildbuddy-io/buildbuddy/proto/scheduler"
 )
 
 const (
 	testSizeEnvVar = "TEST_SIZE"
+	// A BuildBuddy Compute Unit is defined as 1 cpu and 2.5GB of memory.
+	estimatedComputeUnitsPropertyKey = "EstimatedComputeUnits"
+	maxComputeUnits                  = 8         // We can't schedule tasks larger than 8 compute units.
```
Comment (tylerwilliams) [inline enterprise/server/tasksize/tasksize.go:16]: since we already compute the CPU max by looking at the machine -- can we remove this max?

**Review by tylerwilliams — CHANGES_REQUESTED** [review]

File: enterprise/server/tasksize/tasksize.go (hunk 1 above)
Comment (siggisim) [reply enterprise/server/tasksize/tasksize.go]: Done

File: enterprise/server/tasksize/tasksize.go (hunk 2 above)
Comment (siggisim) [reply enterprise/server/tasksize/tasksize.go]: Done - also made the scheduler return an error if trying to schedule a task larger than any of the nodes can handle.

**Review by tylerwilliams — APPROVED** [review]
LGTM

**Review by vadimberezniker — APPROVED** [review]
Can you update AddConnectedExecutor() to also populate the new fields in executionNode?
It doesn't matter for your PR but it would be good to be consistent.

**Comment by siggisim** [comment]
> Can you update AddConnectedExecutor() to also populate the new fields in executionNode?
> It doesn't matter for your PR but it would be good to be consistent.

Done

**Pull request merged**